### PR TITLE
Add/GitHub actions

### PR DESCRIPTION
**Pull request by jeffpaul**
### Description of the Change

Adds linting action, action to deploy new releases to WP.org, and action to deploy changes to readme.txt and/or assets to WP.org.

### Alternate Designs

n/a

### Benefits

automates plugin management tasks

### Possible Drawbacks

Readme/asset changes only to `trunk` or doing a release will trigger the deploy actions, but both should fail with no match on the WP.org SVN side (for now).

### Verification Process

Waiting for PR to run thru checks to see what needs to be fixed before this can be merged.

### Checklist:

- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests passed.

### Applicable Issues

n/a

### Changelog Entry

```
### Added
- GitHub Actions (props @jeffpaul)
```

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    steps:
+    - uses: actions/checkout@v2
+    - name: install node v12
+      uses: actions/setup-node@v1
+      with:
+        node-version: 12
+    - name: npm install
+      run: npm install
+    - name: eslint
+      uses: icrawl/action-eslint@v1
```
Comment (dinhtungdu) [inline .github/workflows/lint.yml:25]:
```suggestion
      uses: icrawl/action-eslint@v1
      with:
        custom-glob: assets/js
```

**Review by dinhtungdu — CHANGES_REQUESTED** [review]
Linting actions are failing, we need to update the composer lock file and specify the JS folder.

**Review by dinhtungdu — DISMISSED** [review]
The PHPCS job is failing due to the outdated `composer.lock` file, @jeffpaul can you please update it? After that, this is ready to be merged

**Comment by jeffpaul** [comment]
@dinhtungdu looks correct now, minus some actual PHPCS errors, yeah?

**Comment by jeffpaul** [comment]
Looks like those errors are all docs related, so I'm good merging this on and adding those in a subsequent PR.

**Pull request merged**